### ISSUE TITLE
RS1024 when calling GroupBy with IEqualityComparer

### ISSUE THREAD

**Issue by YairHalberstadt** [dotnet/roslyn#49385]
**Version Used**: Microsoft.CodeAnalysis.Analyzers 3.3.1

**Steps to Reproduce**:

```csharp
Array.Empty<ISymbol>().GroupBy(x => x.ContainingType, SymbolEqualityComparer.Default);
```

**Expected Behavior**: No error

**Actual Behavior**: RS1024 Compare Symbols Correctly
